OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/base/CryptoOperationFragment.java

```diff
@@ -18,217 +18,104 @@
 
 package org.sufficientlysecure.keychain.ui.base;
 
-import android.app.Activity;
-import android.app.ProgressDialog;
 import android.content.Intent;
-import android.os.Bundle;
-import android.os.Message;
-import android.os.Messenger;
 import android.os.Parcelable;
 import android.support.v4.app.Fragment;
 
-import org.sufficientlysecure.keychain.R;
-import org.sufficientlysecure.keychain.operations.results.InputPendingResult;
 import org.sufficientlysecure.keychain.operations.results.OperationResult;
-import org.sufficientlysecure.keychain.service.KeychainNewService;
-import org.sufficientlysecure.keychain.service.KeychainService;
-import org.sufficientlysecure.keychain.service.ServiceProgressHandler;
 import org.sufficientlysecure.keychain.service.input.CryptoInputParcel;
-import org.sufficientlysecure.keychain.service.input.RequiredInputParcel;
-import org.sufficientlysecure.keychain.ui.NfcOperationActivity;
-import org.sufficientlysecure.keychain.ui.PassphraseDialogActivity;
-import org.sufficientlysecure.keychain.ui.dialog.ProgressDialogFragment;
-
 
 /**
  * All fragments executing crypto operations need to extend this class.
  */
-public abstract class CryptoOperationFragment <T extends Parcelable, S extends OperationResult>
+public abstract class CryptoOperationFragment<T extends Parcelable, S extends OperationResult>
         extends Fragment {
 
-    public static final int REQUEST_CODE_PASSPHRASE = 0x00008001;
-    public static final int REQUEST_CODE_NFC = 0x00008002;
+    private CryptoOperationHelper<T, S> mOperationHelper;
 
-    private void initiateInputActivity(RequiredInputParcel requiredInput) {
+    public CryptoOperationFragment() {
+        // this is implemented here instead of by the fragment so that the corresponding methods in
+        // CryptoOperationFragment may continue using the "protected" modifier.
+        CryptoOperationHelper.Callback callback = new CryptoOperationHelper.Callback<T, S>() {
```

Comment:
It was so that the implementing subclasses wouldn't require any change (`protected` would have to change to `public` - trivial). I don't think that's necessary anymore. I've fixed it in https://github.com/open-keychain/open-keychain/commit/51d3daeccdc697d948c597931d804d738132b8ed.